Refine 'gaining commit privileges'. (#969)
python/devguide: core-developers/become-core-developer.rst

Gaining Commit Privileges
=========================

After a candidate has demonstrated consistent contributions, commit privileges
are granted through these steps:

#. A core developer (submitter, usually the mentor) starts a poll in the

>    `Committers category`_ on the `Python Discourse`_ and cross-post to

Reviewer: ```suggestion
   `Committers category`_ on the `Python Discourse`_ and cross-posts it to
```

Fix grammar

>    `Committers category`_ on the `Python Discourse`_ (``discuss.python.org``).

Reviewer 2: People often confuse Discourse with Discord, so I think the URL is there to prevent/mitigate confusion.

>    `Committers category`_ on the `Python Discourse`_ (``discuss.python.org``).

Reviewer: Yeah, that's why I kept it there, to avoid any ambiguity and follow convention elsewhere. But we could unlink `Python Discourse`, since that URL is arguably somewhat redundant given the category was already linked directly; I only added it originally since I already had the link target defined and that was how it was presented elsewhere.

   - Open for 7 days
   - Results shown upon close

>    - Cross-post the announcement to the `python-committers mailing list`_

Reviewer 2: This item doesn't fit in this list IMHO, since it's a "todo" whereas the other two are info about the poll.  I would move it to the previous sentence ("start a poll in ... and cross-post it to ...").

Author: fixed in 9166247.

#. If the candidate receives at least two-thirds positive votes when the poll closes

>    (as per :pep:`13`), the submitter emails the `steering council

Reviewer: ```suggestion
   (as per :pep:`13`), the submitter `emails the steering council
```

Clarify that this specifically points to the SC email, as opposed to just the Steering Council more generally.

> 2. After the poll has closed, the submitter confirms the membership per
>    :pep:`13` (2/3 of the vote).
> 3. The submitter then emails the
>    `steering council <mailto:steering-council@python.org>`_ with the
>    candidate's email address requesting that the council either accept or reject the proposed
>    membership.

Reviewer 2: ```suggestion
2. If the candidate receives at least two-thirds positive votes when the poll closes
   (as per :pep:`13`), the submitter emails the `steering council
   <mailto:steering-council@python.org>`_ with the candidate's email address
   requesting that the council either accept or reject the proposed membership.
```
Here you could merge the two points.

Reviewer: Also, either the numbering will need to be adjusted, or `#.` auto-numbering used

> 4. Assuming the steering council does not object, a member of the council or delegate

Reviewer 2: ```suggestion
3. Assuming the steering council does not object, a member of the council or delegate
```

Reviewer: Or, `#.` auto-numbering can be used to avoid this going forward

   (approver) will email the candidate:

   - A request for account details as required by

>      `🔒 python/voters <https://github.com/python/voters>`_

Reviewer 3: This "🔒 python/voters" contraction looks good 👍 

GitHub does similar: https://github.com/python/voters

   - A request for the committer's preferred address for subscription to

>      ``python-committers@``

Reviewer 3: I think this would be better written out (linked or not) than using code formatting:

```suggestion
     the python-committers mailing list
```
Or:
```suggestion
     the `python-committers mailing list`_
```

# Before

![image](https://user-images.githubusercontent.com/1324225/202702193-cf4b6cf0-af43-41c6-b830-6e56b29fa422.png)

# After

![image](https://user-images.githubusercontent.com/1324225/202702370-b1484aad-3a79-431d-9794-4485be43137e.png)

Or:

![image](https://user-images.githubusercontent.com/1324225/202702501-d87b30ca-f56d-4e02-974a-28ad36809ee3.png)

Reviewer: Definite +1 on the second option

     to the PSF Conduct WG

#. Once the candidate has provided the pertinent details, the approver will:

>    - enable various new privileges;

Reviewer: ```suggestion
   - enable the various new privileges;
```

Grammar/semantics

>    - enable various new privileges;

Reviewer: Ideally, this would be a bit more specific, although I guess approvers are assumed to be sufficiently experienced to just know this information...

> 5. Once the candidate has provided the pertinent details, the approver will enable
>    various new privileges.
> 6. The approver will add the new committer's details to
>    `🔒 python/voters <https://github.com/python/voters>`_.
> 7. The approver will update the devguide to publicly list the new committer's team
>    membership at :ref:`developers`.
> 8. The approver sends an announcement email to the Committers Discourse category.

Reviewer 2: This can also be simplified a bit:
```suggestion
4. Once the candidate has provided the pertinent details, the approver will:
   - enable various new privileges;
   - add the new committer's details to
     `🔒 python/voters <https://github.com/python/voters>`_;
   - update the devguide to publicly list the new committer's team membership
     at :ref:`developers`;
   - send an announcement email to the Committers Discourse category.
```
